railties/lib/rails/commands/rake/rake_command.rb

```diff
@@ -15,12 +15,11 @@ def printing_commands
         def perform(task, args, config)
           require_rake
 
-          ARGV.replace([task, *args]) # set up ARGV for Rake
-
-          Rake.application.standard_exception_handling do
-            Rake.application.init("rails")
-            Rake.application.load_rakefile
-            Rake.application.top_level
+          Rake.with_application do |rake|
+            load "rails/tasks.rb"
```

Comment:
It's unfortunate that this file has to be loaded each time, but I could not find a sanctioned way to copy loaded tasks from one instance to another.  Neither [`Rake::Application`](https://github.com/ruby/rake/blob/4fe73ff6b19ea1d8490b0442c75fc3a53815c4cf/lib/rake/application.rb) nor [`Rake::TaskManager`](https://github.com/ruby/rake/blob/4fe73ff6b19ea1d8490b0442c75fc3a53815c4cf/lib/rake/task_manager.rb) seem to have the necessary public accessors.

Reply:
Will reloading throw any constant redefined or similar warnings? Is it tough on performance?

Not sure if we know someone involved with Rake we could tap to help fit an API for this.

Reply:
There was one constant redefined warning that I fixed: https://github.com/rails/rails/pull/39137/files#diff-94b3f444d3f09e55370aa7e94f440408R6

As for performance, I'm sure it is slower, but I didn't really notice the difference when I was repeatedly running tasks to try out the fix.  And it should still beat shelling out and loading the entire Rails context in another process, which is what was happening before adding `inline: true`.

It would be cool if Rake supported this use case explicitly.  Depending on how common this use case is, an option like `Rake.with_application(preserve_tasks: true)` would be pretty slick.  Otherwise, something like `Rake::Application#dup_without_top_level` would be next best thing.